### ISSUE TITLE
nanoCUL v3 with CC11010 outputs `c1 is not supported for this cul device` on almost any firmware I've tested so far

### ISSUE THREAD

**Issue by boesing** [wmbusmeters/wmbusmeters#895]
### Type of request

Question

### OS version

macOS

### wmbusmeters version

master branch

### Your message goes here

Hey there,

I've already read plenty of blog posts found via google, read the [wmbusmeters-wiki](https://github.com/wmbusmeters/wmbusmeters-wiki/blob/master/nanoCUL.md), requested support from the shop I've bought my nanoCUL from and installed a bunch of firmwares I've found around the internet:

- the one from the wiki
- the `every-culfw`
- the one from this comment https://github.com/weetmuts/wmbusmeters/issues/45#issuecomment-733601146
- pretty sure I've also tested others

Some of those firmwares did not provided me with any useful information (besides the fact that `wmbusmeters` told me: `Expected cul on /dev/tty.usbserial-A50285BI but did not find it! Ignoring tty!`).

The other firmwares got me to: 

> (wmbus device) timeout request id 1
> (cul) received ""(cul) setting link mode(s) c1 is not supported for this cul device!

I have absolutely no idea what I am doing wrong at this point. The shop which provided me the nanoCUL told me that it is able to handle both `t1` and `c1` mode but I can't run `wmbusmeters` with any of these. I have no idea why they are not able to provide me with the appropriate firmware (I guess they've installed one for FHEM) but they said I have to request support from this project. :-/ 

So now I'm stuck and have no idea on how to run my `nanoCUL v3`.

This is the output from `wmbusmeters` when running the nanoCUL with the firmware provided from the wiki (i.e. the one with "bufsize300" - `nanoCUL_r568_mbus_c1t1_bufsize300.hex`):

```
→ ./wmbusmeters --debug --logtelegrams --verbose /dev/tty.usbserial-A50285BI:cul:c1
(wmbusmeters) version: _
(config) using device: /dev/tty.usbserial-A50285BI:cul:c1 
(config) number of meters: 0
(lookup) with file/hex "/dev/tty.usbserial-A50285BI" cul
(serial) check if /dev/tty.usbserial-A50285BI can be accessed
(serial) tty /dev/tty.usbserial-A50285BI can be accessed
(serial) EVENT thread interrupted
(serial) before config: /dev/tty.usbserial-A50285BI speed(B9600) input() output() control(CREAD|CSIZE|HUPCL) local() special_chars(4,255,255,127,23,21,18,255,3,28,26,25,17,19,22,15,1,20,255) modem(DTR|RTS)
(serial) after config:  /dev/tty.usbserial-A50285BI speed(B38400) input() output() control(CLOCAL|CREAD|CSIZE) local() special_chars() modem(DTR|RTS)
(serialtty) opened /dev/tty.usbserial-A50285BI fd 3 (detect cul)
(serial /dev/tty.usbserial-A50285BI) sent "560D0A"
(serial) EVENT thread interrupted
(serial) received binary ""
(cul) probe response ""
(serial /dev/tty.usbserial-A50285BI) sent "560D0A"
(serial) EVENT thread interrupted
(serial) received binary ""
(cul) probe response ""
(serial /dev/tty.usbserial-A50285BI) sent "560D0A"
(serial) EVENT thread interrupted
(serial) received binary "5620312E3637206E616E6F43554C3836385F723536380D0A"
(cul) probe response "V 1.67 nanoCUL868_r568<0D><0A>"
(serialtty) closed /dev/tty.usbserial-A50285BI (detect cul)
(cul) are you there? yes
(main) opening /dev/tty.usbserial-A50285BI:cul:c1
(serial) EVENT thread interrupted
Started config cul on /dev/tty.usbserial-A50285BI listening on c1
(cul) on /dev/tty.usbserial-A50285BI
(serial) EVENT thread interrupted
(serial) before config: /dev/tty.usbserial-A50285BI speed(B9600) input() output() control(CREAD|CSIZE|HUPCL) local() special_chars(4,255,255,127,23,21,18,255,3,28,26,25,17,19,22,15,1,20,255) modem(DTR|RTS)
(serial) after config:  /dev/tty.usbserial-A50285BI speed(B38400) input() output() control(CLOCAL|CREAD|CSIZE) local() special_chars() modem(DTR|RTS)
(serialtty) opened /dev/tty.usbserial-A50285BI fd 3 (cul)
(main) regular reset of cul  on /dev/tty.usbserial-A50285BI will happen every 82800 seconds
(cul) set link mode c
(serial /dev/tty.usbserial-A50285BI) sent "6272630D0A"
(serial) EVENT thread interrupted
(wmbus device) timeout request id 1
(cul) received ""(cul) setting link mode(s) c1 is not supported for this cul device!
(serial) stopping manager
```

I have flashed that firmware with `avrdude` which seems the only software wich is able to patch my nanoCUL v3:

```
→ avrdude -p atmega328p -P /dev/tty.usbserial-A50285BI -b 115200 -c arduino -U flash:w:nanoCUL_r568_mbus_c1t1_bufsize300.hex:i

avrdude: AVR device initialized and ready to accept instructions

Reading | ################################################## | 100% 0.01s

avrdude: Device signature = 0x1e950f (probably m328p)
avrdude: NOTE: "flash" memory has been specified, an erase cycle will be performed
         To disable this feature, specify the -D option.
avrdude: erasing chip
avrdude: reading input file "nanoCUL_r568_mbus_c1t1_bufsize300.hex"
avrdude: writing flash (12320 bytes):

Writing | ################################################## | 100% 4.65s

avrdude: 12320 bytes of flash written
avrdude: verifying flash memory against nanoCUL_r568_mbus_c1t1_bufsize300.hex:

Reading | ################################################## | 100% 4.17s

avrdude: 12320 bytes of flash verified

avrdude done.  Thank you.
```

----

Can you guys help me with this problem?
Regards,
Max

**Comment by TomTomNavigator** [comment]
Hey Max,

had a similar issue, I blogged about it here: https://thomasheinz.net/setting-link-modes-c1-is-not-supported-for-this-cul-device/

Seems like you also tried that already, I just wanted to leave it here - just in case.
I will soon get another nanocul - I report back how it goes!

**Comment by boesing** [comment]
Hey Tom,
I indeed found your blog post and tried this already. It feels as I've installed almost every firmware I could find - none seems to work. I dont understand why this is the case tho... Not sure what actually causes that message as well. Is it possible that the CC11010 is not properly soldered? 
I am lacking the skills to verify this as I bought it from Amazon... It looks professional to me tho. 

Bought this one: 

nanoCUL USB Stick FTDI CC1101... https://www.amazon.de/dp/B0BG3JD6FV

**Comment by TomTomNavigator** [comment]
Hi Max,

I also had to a lot of research in the beginning, I am using a busware cul v3 though - which I bought via ebay kleinanzeigen.

I also ordered another nanocul (for backup / neighbor) - these are based on Arduino; if I am not mistaken; and you can also change the bootloader to optiboot. That would be at least my next troubleshooting step - if mine would not work with c1.

**Comment by TomTomNavigator** [comment]
I received the nanocul. This one still has the old bootloader on it, that's why I only could write it with 57600 baud.

`avrdude -D -p atmega328p -P /dev/ttyUSB0 -b 57600 -c arduino -U flash:w:nanoCUL_r568_mbus_c1t1_bufsize300.hex`


Testing:
```
wmbusmeters auto:c1                                                                                                                                                                                                  Started auto cul on /dev/ttyUSB0 listening on c1
No meters configured. Printing id:s of all telegrams heard!
Received telegram from: 24326566
          manufacturer: (TCH) Techem Service (0x5068)
                  type: Smoke detector (0xf0)
                   ver: 0x76
                device: cul
                  rssi: -84 dBm
                driver: tsd2
Received telegram from: 24362909
          manufacturer: (TCH) Techem Service (0x5068)
                  type: Smoke detector (0xf0)
                   ver: 0x76
                device: cul
                  rssi: -87 dBm
                driver: tsd2
[trimmed]
```

**Comment by spandi** [comment]
@boesing : I have the very same issue on Mac. Also tried various firmwares but I'm also stuck with the same error:
```
...
(serial) before config: /dev/cu.usbserial-A50285BI speed(B9600) input() output() control(CREAD|CSIZE|HUPCL) local() special_chars(4,255,255,127,23,21,18,255,3,28,26,25,17,19,22,15,1,20,255) modem(DTR|RTS)
(serial) after config:  /dev/cu.usbserial-A50285BI speed(B38400) input() output() control(CLOCAL|CREAD|CSIZE) local() special_chars() modem(DTR|RTS)
(serialtty) opened /dev/cu.usbserial-A50285BI fd 3 (cul)
(main) regular reset of cul  on /dev/cu.usbserial-A50285BI will happen every 82800 seconds
(cul) set link mode c
(serial /dev/cu.usbserial-A50285BI) sent "6272630D0A"
(serial) EVENT thread interrupted
(wmbus device) timeout request id 1
(cul) received ""(cul) setting link mode(s) c1 is not supported for this cul device!
```

Did you ever solve this issue?

**Comment by boesing** [comment]
It actually worked on my raspberry pi back in the days. thats arm based as well, I just guess that there is something else going on when running on mac. In the end, it turned out that I still lack the decryption key of my provider and thus did not continue working on it.

**Comment by spandi** [comment]
@boesing : thanks a lot for your feedback which was very helpful. I'm actually using a Raspberry, but I did not get nanoCUL to work with it, so I wanted to test the setup on my Mac. Now I know that it does not work there either.

On the Raspberry I get
```
(serialtty) closed /dev/ttyUSB0 (detect cul)
(cul) are you there? no
```
It used to work with an older CUL device.

When I test the nanoCUL with
```
screen /dev/ttyUSB0 38400
```
and type 'V' I correctly get:
```
V 1.67 nanoCUL868
```
I also use the correct firmware ([nanoCUL868-1.67.hex](https://cdn.shopify.com/s/files/1/0761/4237/9347/files/nanoCUL868-1.67.hex?v=1695996166)). Anyway, that's a different issue than this one.

BTW: I have exactly the same issue with my provider not giving me the AES decryption key because it's a shared key 😞. As we anyway had to changes some pipes, I got my own meter now and it used to work fine for some month, but then (presumably) a Raspberry or wmbusmeters update or hardware issue on my old CUL device broke my working setup 😞.

**Comment by spandi** [comment]
I was able to flash my older CUL V3.2 device again and tada, it started to work on my Mac. So it definitely seems to be an issue with nanoCUL and wmbusmeters on Mac.

However, I did not get the CUL V3.2 or the nanoCUL working on the Raspberry. It does not properly detect it. I will create a separate issue for this.

**Comment by spandi** [comment]
If someone ends up here: I spent endless hours trying to get the CUL and nanoCUL working on the Raspberry, but without success. I was still using bullseye (latest updates installed) and it seem that the USB port could be opened at first but not reopened, so wmbusmeters was not working properly.

I finally got it working with a fresh installation (using bookworm this time). Now the CUL receives telegrams. The nanoCUL looks fine as well from a wmbusmeters point of view, but it does not receive anything. I suspect some hardware issue with the RF part of the nanoCUL. Anyway, I will give up on the nanoCUL now as I have a working setup with the CUL V3.2.
